File: Lab 1/Python/MatrixMul.py
import time

Matrix_A=[]
Matrix_B=[]
Matrix_C=[]

print("Please enter x_A,y_A,x_B,y_B")
x_A,y_A,x_B,y_B = map(int, input().split())

if (y_A != x_B)or(x_A == 0)or(y_A == 0)or(x_B == 0)or(y_B == 0):
    print("Error")
else:
    print("Please enter Matrix_A")

    for i in range(x_A):
        Matrix_A.append([])
        temp=input().split()
        for j in range(y_A):
            Matrix_A[i].append(int(temp[j]))
    #print(Matrix_A)

    print("Please enter Matrix_B")

    for i in range(x_B):
        Matrix_B.append([])
        temp = input().split()
        for j in range(y_B):
            Matrix_B[i].append(int(temp[j]))
    #print(Matrix_B)

    start_time = time.time()
    for i in range(x_A):
        Matrix_C.append([])
        for j in range(y_B):
            Matrix_C[i].append(0)
            for k in range(y_A):
                Matrix_C[i][j]+=Matrix_A[i][k]*Matrix_B[k][j]

    end_time=time.time()

    print("Matrix_C:")
    print(Matrix_C)
    print("time = ",end_time-start_time," s")

File: Lab 1/Java/MatrixMul.java
import java.util.Scanner;

public class MatrixMul {

    public static void main(String[] args) {

        int Matrix_A[][] = new int[100][100];
        int Matrix_B[][] = new int[100][100];
        int Matrix_C[][] = new int[100][100];

        int x_A,y_A,x_B,y_B;

        System.out.println("Please enter x_A,y_A,x_B,y_B");

        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine();
        String ss[] = s.split(" ");
        x_A=Integer.parseInt(ss[0]);
        y_A=Integer.parseInt(ss[1]);
        x_B=Integer.parseInt(ss[2]);
        y_B=Integer.parseInt(ss[3]);


        System.out.println("Please enter Matrix_A");
        for(int i = 0;i<x_A;i++)
        {
            Scanner sc_A = new Scanner(System.in);
            String s_A = sc_A.nextLine();
            String ss_A[] = s_A.split(" ");
            for(int j = 0;j<y_A;j++)
            {
                Matrix_A[i][j] = Integer.parseInt(ss_A[j]);
            }
        }

        System.out.println("Please enter Matrix_B");
        for(int i = 0;i<x_B;i++)
        {
            Scanner sc_B = new Scanner(System.in);
            String s_B = sc_B.nextLine();
            String ss_B[] = s_B.split(" ");
            for(int j = 0;j<y_B;j++)
            {
                Matrix_B[i][j] = Integer.parseInt(ss_B[j]);
            }
        }

        long start_time = System.currentTimeMillis();
        for(int i = 0;i<x_A;i++)
        {
            for(int j = 0;j<y_B;j++)
            {
                Matrix_C[i][j]=0;
                for(int k = 0;k<y_A;k++)
                {
                    Matrix_C[i][j] += Matrix_A[i][k]*Matrix_B[k][j];
                }
            }
        }

        long end_time = System.currentTimeMillis();
        System.out.println("time: " + (end_time-start_time) + " ms");

        System.out.println("Matrix_C");
        for(int i = 0;i<x_A;i++)
        {
            for(int j = 0;j<y_B;j++)
            {
                System.out.printf("%d",Matrix_C[i][j]);
                if(j == y_B-1)
                {
                    System.out.println();
                }
                else
                {
                    System.out.print(",");
                }
            }
        }
    }
}